dlio_benchmark/data_loader/torch_data_loader.py

```diff
@@ -44,11 +46,14 @@ def __init__(self, format_type, dataset_type, epoch, num_samples, num_workers, b
         self.reader = None
         self.num_images_read = 0
         self.batch_size = batch_size
+        args = ConfigArguments.get_instance()
+        self.serial_args = pickle.dumps(args)
         if num_workers == 0:
             self.worker_init(-1)
 
     @dlp.log
     def worker_init(self, worker_id):
+        pickle.loads(self.serial_args)
         logging.debug(f"{utcnow()} worker initialized {worker_id} with format {self.format_type}")
```

Comment:
How would this update the singleton? Won’t we have to have a static method in config to make sure we update the config with this loaded instance on the init of worker. 

Also can u add a new torch_spawn_data_loader.py class which add the multiprocessing context to the data loader as spawn and that to tests to make sure it all works.

Reply:
When pickle.loads() is called, pickle sees that the ConfigArguments class now has a __setstate__ method.  That causes pickle to populate the memory-allocated class object with all the fields in the pickle, but it does NOT call the ConfigArguements __init__ method.  The __setstate__ method then sets ConfigArguments.__instance to the newly created/unpickled ConfigArguments class, so any code which references ConfigArguments will now always get the new unpickled class.

Reply:
I am working on tests.

Reply:
@krehm I added the tests to github actions. Let us see whether it works.

Reply:
@krehm @hariharan-devarajan looks like it does not work. Please see the github action runs.